src/libs/sfdk/virtualmachine.cpp

```diff
@@ -372,6 +372,23 @@ int VirtualMachine::storageSizeMb() const
     return d->virtualMachineInfo.storageSizeMb;
 }
 
+int VirtualMachine::freeStorageSizeMb() const
+{
+    Q_D(const VirtualMachine);
+    QTC_ASSERT(d->initialized(), return -1);
+    QTC_ASSERT(d->features & ReserveStorageSize, return -1);
+    const Utils::FilePath dfPath = Utils::FilePath::fromString(d->cachedInfo().sharedConfig)
+            .pathAppended("df.cache");
+    FileReader reader;
+    if (!reader.fetch(dfPath.toString(), QIODevice::ReadOnly))
```

Comment:
We should say something qCritical here and below.

Reply:
sure